### PR TITLE
feat: add better support for research-bots

### PR DESCRIPTION
**Pull request by daniel1302**
# Changes

- Added support for perpetual markets
- Removed a few unused imports
- Added support for the mainnet mirror network 
- Move price feed to another feed and wait for feed to avoid having price = None
- Allow to pass more Agents instances to have more than 3 traders in the bots.

New revision of changes from https://github.com/vegaprotocol/vega-market-sim/pull/481

# Testing

Triggered build here: https://jenkins.vega.rocks/job/common/job/vega-market-sim/2317/console

### CODE REVIEW

**Review by cdummett — DISMISSED** [review]

File: vega_sim/api/data.py [hunk 1]
```diff
@@ -946,9 +946,21 @@ def positions_by_market(
                 market_info = data_raw.market_info(
                     market_id=pos.market_id, data_client=data_client
                 )
+
+            settlement_asset_id = (
+                market_info.tradable_instrument.instrument.future.settlement_asset
+            )
+            if len(settlement_asset_id) < 1:
```
Comment (TomMcL) [inline vega_sim/api/data.py]: This is an odd check, can we just do `if not settlement_asset_id` ?

File: vega_sim/scenario/common/utils/price_process.py [hunk 2]
```diff
@@ -217,9 +231,24 @@ def __next__(self):
...
+        while self.latest_price is None:
+            time.sleep(0.33)
         return self.latest_price
 
 
+_live_prices = {}
+_live_prices_lock = threading.Lock()
+
+
+def getLivePrice(product: str) -> LivePrice:
```
Comment (TomMcL) [inline vega_sim/scenario/common/utils/price_process.py]: `get_live_price`

Comment (daniel1302) [reply vega_sim/scenario/common/utils/price_process.py]: Renamed

File: vega_sim/api/data.py (hunk 1 above)
Comment (daniel1302) [reply vega_sim/api/data.py]: Fixed this + 2 other places with similar conditions

**Comment by daniel1302** [comment]
@TomMcL triggered another run after fixing your comments: https://jenkins.vega.rocks/job/common/job/vega-market-sim/2362/console

**Review by TomMcL — APPROVED** [review]

**Pull request merged**